### PR TITLE
Update calls to OrtArenaCfg constructor to pass additional parameter.

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Update calls to OrtArenaCfg constructor to pass additional parameter.

### Motivation and Context

Updating some call sites after change in #15983.
Fix CI build.

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: onnxruntime/core/providers/cann/cann_execution_provider.cc [hunk 1]
```diff
@@ -1455,6 +1455,7 @@ void CANNExecutionProvider::RegisterAllocator(AllocatorManager& allocator_manage
                                                         static_cast<int>(info_.arena_extend_strategy),
                                                         -1,
                                                         -1,
+                                                        -1,
                                                         -1)},
```
Comment (tianleiwu) [inline onnxruntime/core/providers/cann/cann_execution_provider.cc:1459]: last one is int64:  -1L

Comment (edgchen1) [reply onnxruntime/core/providers/cann/cann_execution_provider.cc:1459]: the conversion should be fine from int to int64_t

Comment (edgchen1) [reply onnxruntime/core/providers/cann/cann_execution_provider.cc:1459]: I'll mark this as resolved - let me know if you want me to update it

**Review by tianleiwu — APPROVED** [review]

**Review by cloudhan — APPROVED** [review]
I can locally verify this fixed the problem.

**Pull request merged**